joschahenningsen/TUM-Live
# Add Course REST Endpoint

### Motivation and Context
In preparation of the course page changes [#791](https://github.com/joschahenningsen/TUM-Live/pull/791) and maybe even [this discussion](https://github.com/joschahenningsen/TUM-Live/discussions/845) add a REST endpoint which provides
all information of a given course. 

Additionally, split the courses router into a general and an admin router since the file and router definition slowly but surely 
got bigger and bigger.

This endpoint can then be used to filter for VOD's, Planned, and Live streams in Typescript and clean up the go-template mess I made in 2021.

❓We might need to discuss what attributes we want to hide with ``json:"-"`` or ``json:",omitempty"``. 

### Changes 

Regarding the changes, admin course routes can now be found in `courses_admin.go` and the associated tests 
in `courses_admin_test.go`. That's why one file is missing 1000+ lines and two new files are added.

### Description
Add `/api/courses/:id` endpoint.

### Steps for Testing
1. Run all `api` tests
2. Curl or Postman to endpoint (e.g. `localhost:8081/api/courses/262`)
3. Log-In and in the browser console run: `fetch("/api/courses/262").then((res) => res.json()).then((json) => console.log(json))`
    - Now the watched state should be in the response too. 
    - The person updating the frontend might want to remove this part of the handler function since it can easily be created in Typescript. For the month name one can use 
        ```javascript
      const month = new Date('2022-11-29T12:00:00+01:00').toLocaleString('default', { month: 'long' })
